api/types/system/system.go

```diff
@@ -0,0 +1,11 @@
+package system // import "github.com/docker/docker/api/types/system"
+
+// ListenerInfo provides information about an address that the API is listening on.
+type ListenerInfo struct {
+	// Address is the address that the daemon API is listening on.
+	Address string
+	// Insecure indicates if the address is insecure, which could be either
+	// if the daemon is configured without TLS client verification, or if
+	// TLS verify is disabled.
+	Insecure bool
```

Comment:
Maybe break down `Address` into a `Type` (`tcp`, `unix-socket`, `named-pipe`) and an address.
Then add an extra field to show the config like:

```go
type ListenerInfo {
    Type string
    Address string
    TCPConfig *ListenerTCPConfig `json:"tcp-config,omitempty"`
}

type ListenerTCPConfig {
    Encrypted bool
    Authenticated bool
}
```